kview/src/main/java/dev/ikm/komet/kview/mvvm/view/pattern/PatternFieldsController.java

```diff
@@ -157,24 +164,35 @@ private void initialize() {
         });
 
         IntegerProperty totalExistingfields = patternFieldsViewModel.getProperty(TOTAL_EXISTING_FIELDS);
-        SimpleIntegerProperty fieldOrderProp = patternFieldsViewModel.getProperty(FIELD_ORDER);
+        // This had to be changed to ObjectProperty<Integer>. Else give a runtime casting exception.
+        ObjectProperty<Integer> fieldOrderProp = patternFieldsViewModel.getProperty(FIELD_ORDER);
         ObservableList<Integer> fieldOrderOptions = patternFieldsViewModel.getObservableList(FIELD_ORDER_OPTIONS);
 
         StringProperty displayNameProp = patternFieldsViewModel.getProperty(DISPLAY_NAME);
         ObjectProperty<ConceptEntity> dataTypeProp = patternFieldsViewModel.getProperty(DATA_TYPE);
         ObjectProperty<ConceptEntity> purposeProp = patternFieldsViewModel.getProperty(PURPOSE_ENTITY);
         ObjectProperty<ConceptEntity> meaningProp = patternFieldsViewModel.getProperty(MEANING_ENTITY);
 
-        fieldOrderProp.bind(fieldOrderComboBox.getSelectionModel().selectedItemProperty());
         displayNameTextField.textProperty().bindBidirectional(displayNameProp);
         dataTypeComboBox.valueProperty().bindBidirectional(dataTypeProp);
         fieldOrderComboBox.setItems(fieldOrderOptions); // Set the items in fieldOrder
-
+        fieldOrderComboBox.valueProperty().bindBidirectional(fieldOrderProp);
 
         displayNameProp.addListener(fieldsValidationListener);
         dataTypeProp.addListener(fieldsValidationListener);
-        purposeProp.addListener(fieldsValidationListener);
-        meaningProp.addListener(fieldsValidationListener);
+        purposeProp.addListener((obs, oldVal, newVal) -> {
+            if(newVal != null){
+                addPurposeToForm(purposeProp.getValue());
+            }
+            fieldsValidationListener.changed(obs, oldVal, newVal);
+        });
```

Comment:
This should not be combined. It should be the following:
```java
purposeProp.addListener(fieldsValidationListener);
purposeProp.addListener((obs, oldVal, newVal) -> addPurposeToForm(newVal)); 
// the function should do the null check.
```
Same goes for meaning below.

Reply:
I will fix this